### ISSUE TITLE
TSNE on descriptors outputted by SuperPoint?

### ISSUE THREAD

**Issue by hnyll** [cvg/LightGlue#105]
Bascially title. Also I'm not sure how to interpret the data for the descriptors. sum1 help a newbie pls? 
this is all i have but im not sure if in doing it right

`feats0 = extractor.extract(image0.to(device))
feats1 = extractor.extract(image1.to(device))

matches01 = matcher({"image0": feats0, "image1": feats1})
feats0, feats1, matches01 = [
        rbd(x) for x in [feats0, feats1, matches01]
]  # remove batch dimension

kmeans = KMeans(n_clusters=3, random_state=42)
cluster_assignments = kmeans.fit_predict(feats1['descriptors'].cpu())
num_keypoints = feats1['descriptors'].shape[0]
arbitrary_labels = np.arange(num_keypoints)

embedding = TSNE(init='pca',learning_rate='auto', square_distances=True, n_components=2, perplexity=30, metric='cosine').fit_transform(feats1["descriptors"].cpu().numpy().astype(np.float32))
plt.scatter(embedding[:, 0], embedding[:, 1],c=cluster_assignments, cmap='viridis', s=5)
plt.title(f'Descriptors in Embedding Space {im1}')
plt.show()`
